babel/numbers.py

```diff
@@ -957,17 +961,18 @@ def parse_precision(p):
     return NumberPattern(pattern, (pos_prefix, neg_prefix),
                          (pos_suffix, neg_suffix), grouping,
                          int_prec, frac_prec,
-                         exp_prec, exp_plus)
+                         exp_prec, exp_plus, number)
 
 
 class NumberPattern:
 
     def __init__(self, pattern, prefix, suffix, grouping,
-                 int_prec, frac_prec, exp_prec, exp_plus):
+                 int_prec, frac_prec, exp_prec, exp_plus, number_pattern=None):
```

Comment:
Could you add a type hint for the new argument? We haven't had type hints much anywhere yet, but it'll happen sooner or later.

Reply:
I actually just type-hinting pretty much all modules except `messages` and some of `support` that in types-babel. Maybe I could make a PR to copy them all over?